.github/workflows/build-and-test.yml

```diff
@@ -11,9 +11,10 @@ env:
   # Cache downloads for this workflow consistently run in under 1 minute
   SEGMENT_DOWNLOAD_TIMEOUT_MINS: 5
 
+# Do not cancel this workflow on main. See https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16616
 concurrency:
-  group: ${{ github.workflow }}-${{ github.head_ref }}
-  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
```

Comment:
What does the `||` do in this scenario?

Sorry, I don't get to use github actions often

Reply:
It's a fallback value. `github.head_ref` will be used if not nil, `github.run_id` otherwise. More details in the description above.